File: Algorithms/014.LongestCommonPrefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if (strs.size() == 0) {
            return "";
        }
        if (strs.size() == 1) {
            return strs[0];
        }
        int i;
        for (i=0; i < strs[0].size(); i++) {
            int j;
            for (j=1; j < strs.size(); j++) {
                if (i < strs[j].size() && strs[j][i] == strs[0][i]) {
                    ;
                }
                else {
                    break;
                }
            }
            if (j != strs.size()) {
                break;
            }
        }
        return strs[0].substr(0, i);
    }
};

File: Algorithms/py/014.LongestCommonPrefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        if len(strs) == 0:
            return ""
        if len(strs) == 1:
            return strs[0]
        index = 0
        for i in range(len(strs[0])):
            flag = True
            for j in range(1, len(strs)):
                if i < len(strs[j]) and strs[0][i] == strs[j][i]:
                    continue
                else:
                    flag = False
            if not flag:
                break
            index = i+1
        return strs[0][:index]
